plugins/modules/rulebook_activation.py

```diff
@@ -539,6 +597,27 @@ def main() -> None:
         except EDAError as e:
             module.fail_json(msg=f"Failed to delete rulebook activation: {e}")
 
+    if restart:
+        restart_interval = module.params.get("restart_interval")
+        restart_wait_time = module.params.get("restart_wait_time")
+        if module.params.get("restart_wait"):
+            end_time = time.time() + restart_wait_time
+            result = restart_activation(activation, module, controller)
+            time.sleep(10)
+            while time.time() < end_time:
+                activation = update_activation(activation, module, controller)
+                if (
+                    activation["status"] == "running"
+                    or activation["status"] == "completed"
+                ):
+                    break
+                time.sleep(restart_interval)
+            result["msg"] = f"Current Activation details: {activation}"
```

Comment:
What is the purpose of waiting after restart? Isn't that to make sure the activation is restarted and back to running state? So shouldn't we exit the module with error if we cannot detect this status within the timeout window? But there might be cases that the timeout is set too short and the activation still reaches the desired status after timeout.

Reply:
Agree @bzwei The timeout is set by the playbook author and if it does not reach the expected state in the time window it should fail. This a convenience feature to avoid to the user to write their own loop and it is aligned with other modules with similar features like projects in AAP. Projects are cancelled after the timeout but they are one shot actions, so it makes sense, in this case I think it is better just to raise the error. 

For job templates there is [a specific module](https://docs.ansible.com/ansible/latest/collections/awx/awx/job_wait_module.html#ansible-collections-awx-awx-job-wait-module) to do this. We can consider to use a new module in later iterations.

Reply:
It is tricky what to do after timeout. We agree it should fail the module. Because there is no remedy after timeout, the only choice seems just do nothing 😄